process/runner.go

```diff
@@ -165,6 +175,27 @@ func (r *runner) runInitializers(config config.Config) bool {
 	return true
 }
 
+func (r *runner) runFinalizers() bool {
+	return r.unwindInitializers(r.processes.NumInitializers())
+}
+
+func (r *runner) unwindInitializers(beforeIndex int) bool {
+	r.logger.Info("Running finalizers")
+
+	success := true
+	initializers := r.processes.GetInitializers()
+
+	for i := beforeIndex - 1; i >= 0; i-- {
+		if err := r.finalizeWithTimeout(initializers[i]); err != nil {
+			r.errChan <- errMeta{err: err, source: initializers[i]}
```

Comment:
Is there something that's listening to this channel consistently? Is there a chance that this blocks forever?

Reply:
The values from this chan are logged and placed onto outChan, which is the return value of Run (and consumed by Boot). The contract of the runner necessitates reading from this chan to completion.